File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    
    heap = []
    
    for i in scoville :
        heapq.heappush(heap,i)
        
    if heap[0] >= K :
        
        return 0
    
    cnt = 0
    while True :
        
        cnt += 1
        
        if len(heap) == 1 :
            return -1 
        else :
            x = heapq.heappop(heap)
            y = heapq.heappop(heap)
            heapq.heappush(heap,x+2*y)
            
        if heap[0] >= K :
            return cnt
        
        
        
    
        
        
    
    

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*; 

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        int n = scoville.length;
        int cnt = 0 ;
        
        for (int i = 0 ; i<n ; i++)
        {
            pq.add(scoville[i]);
        }
        
        while(pq.size() > 1)
        {
            if (pq.peek() >= K)
            {
                break;
            }
            
            //더 요리해야되면
            cnt +=1;
            int one = pq.poll();
            int two = pq.poll();
            
            pq.add(one+two*2);
        }
        
        if (pq.peek() < K)
        {
            cnt = -1;
        }
        
        return cnt ;
    }
}